mesonbuild/compilers/mixins/gnu.py

```diff
@@ -164,14 +170,14 @@ def get_buildtype_args(self, buildtype: str) -> T.List[str]:
 
     @abc.abstractmethod
     def get_optimization_args(self, optimization_level: str) -> T.List[str]:
-        raise NotImplementedError("get_optimization_args not implemented")
```

Comment:
Why are these exceptions removed?

Reply:
they're redundant with `abc.abstracetmethod`, you can't instantiate a class that has non-overloaded abstract methods, the `ABCMeta` metaclass prevents it.

Reply:
OK, thanks for explaining